### PR TITLE
[CHORE] Switch to react-native-localize

### LINKED ISSUE #899: Switch to react-native-localize
react-native-i18n is [deprecated](https://github.com/AlexanderZaytsev/react-native-i18n#%EF%B8%8F-this-package-is-deprecated) and we need to switch to the [recommended lib from RN community](https://github.com/react-native-community/react-native-localize)

### PR DESCRIPTION
**Pull request by milabressan**
@RocketChat/ReactNative

Closes #899

### CODE REVIEW

**Comment by pranavpandey1998official** [comment]
@milabressan thanks for PR, we don't use npm i can you remove package.json.lock

File: android/.project [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (diegolmello) [inline android/.project]: Can you remove this file from git?

File: android/.settings/org.eclipse.buildship.core.prefs [hunk 2]
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
Comment (diegolmello) [inline android/.settings/org.eclipse.buildship.core.prefs]: Please remove this file as well

File: app/i18n/index.js [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
-import fr from './locales/fr';
-import de from './locales/de';
-import ptBR from './locales/pt-BR';
-import zhCN from './locales/zh-CN';
-import ptPT from './locales/pt-PT';
+import i18n from 'i18n-js';
+import { I18nManager } from 'react-native';
+import * as RNLocalize from 'react-native-localize';
+import {
+	en, ru, fr, de, ptBR, zhCN, ptPT
```
Comment (diegolmello) [inline app/i18n/index.js]: Sort alphabetically

**Comment by milabressan** [comment]
@diegolmello Thanks for the review.
I already made the corrections. :)

**Comment by diegolmello** [comment]
@milabressan Lint is failing: https://circleci.com/gh/RocketChat/Rocket.Chat.ReactNative/9003?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by diegolmello** [comment]
@milabressan Hi. Are you going to fix lint?

File: ios/Podfile [hunk 4]
```diff
@@ -46,7 +46,8 @@ target 'RocketChatRN' do
   pod 'GoogleIDFASupport', '~> 3.14.0'
   pod 'Firebase/Performance', '~> 5.20.1'
 
-  use_unimodules!
```
Comment (diegolmello) [inline ios/Podfile]: Don't remove this line.

File: android/app/build.gradle [hunk 5]
```diff
@@ -172,9 +172,9 @@ android {
 }
 
 dependencies {
-    addUnimodulesDependencies()
```
Comment (diegolmello) [inline android/app/build.gradle]: Don't remove this line.

**Pull request closed without merging**